File: 3168-minimum-number-of-chairs-in-a-waiting-room/3168-minimum-number-of-chairs-in-a-waiting-room.swift
class Solution {
    func minimumChairs(_ s: String) -> Int {
        var largestChair = 0
        var curr = 0
        
        for char in s {
            if char == "E" {
                curr += 1
                
                if curr > largestChair {
                    largestChair = curr
                }
            } else {
                curr -= 1
            }
        }
        
        return largestChair
    }
}

File: 3168-minimum-number-of-chairs-in-a-waiting-room/3168-minimum-number-of-chairs-in-a-waiting-room.rs
impl Solution {
    pub fn minimum_chairs(s: String) -> i32 {
        let mut result = 0;
        let mut curr = 0;

        for c in s.chars().into_iter() {
            if c == 'E' {
                curr += 1;

                if result < curr {
                    result = curr
                }
            } else {
                curr -= 1;
            }
        }

        result       
    }
}
